@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1801,14 +1801,16 @@ export class TextElementsRenderer {
                     numNotVisible++;
                     return false;
                 }
-                const textPath = new THREE.Path();
-                if (screenPoints[1].x - screenPoints[0].x >= 0) {
-                    tempScreenPosition.copy(screenPoints[0]);
-                    for (let i = 0; i < screenPoints.length - 1; ++i) {
-                        textPath.add(new SimpleLineCurve(screenPoints[i], screenPoints[i + 1]));
-                    }
-                } else {
+
+                let textPath = new THREE.Path();
+                tempScreenPosition.copy(screenPoints[0]);
+                for (let i = 0; i < screenPoints.length - 1; ++i) {
+                    textPath.add(new SimpleLineCurve(screenPoints[i], screenPoints[i + 1]));
+                }
+                // Flip the path if the label is gonna be rendered downwards.
+                if (textPath.getPoint(0.5).x - textPath.getPoint(0.51).x > 0) {
```

Comment:
what does 0.5 mean, is this the center of the first path? or the center of the whole line?

Reply:
I know, but the current issue is that just based on our current screen projected points, there's no way to know where the center of the final path will be (since every ``SimpleLineCurve`` might have different lengths). The path provide us the center of all the combined curves, which is why we use it for evaluation.

We could compute the length of every ``SimpleLineCurve`` segments as we project points to the screen, but this would happen for all labels, not only for the flipped ones.